### ISSUE TITLE
Not able to use Advanced margin_padding control multiple times ?

### ISSUE THREAD

**Issue by SatishKumarPrajapati** [elegantthemes/create-divi-extension#125]
### Problem Description

There is no option to use advanced margin padding control to use multiple times as other controls are easy to use multiple times.

### Steps To Reproduce
1. $advanced_fields['margin_padding']['part2'] = [
        'css'             => [
            'main'    => [
                'margin' => "%%order_class%% .bpd-st-transform-text-title .bpd-st-rest-text",
                'padding' => "%%order_class%% .bpd-st-transform-text-title .bpd-st-rest-text",
            ],
         ],
        'hide_text_align' => true,
        'tab_slug'        => 'advanced',
        'toggle_slug'     => 'part2',
 	];

**Comment by etstaging** [comment]
### QA Checklist
<strong>Reporter</strong>
- [ ] Helpful title used &nbsp;<img src="https://assets-cdn.github.com/images/icons/emoji/unicode/274c.png" width="16" height="16"/>
  * ***Helpful title format:*** `Scope(s) :: One Sentence To Describe The Issue`
  * ***Example:*** `VB :: Floating Toolbar :: Jumps To Top Of Page`
- [ ] Detailed description <img src="https://assets-cdn.github.com/images/icons/emoji/unicode/274c.png" width="16" height="16"/>
  * ***Description should usually be 150+ characters.***
- [ ] Steps to reproduce included <img src="https://assets-cdn.github.com/images/icons/emoji/unicode/274c.png" width="16" height="16"/>
- [ ] Support Ticket URL included
- [ ] Image and/or gif

<strong>Developer</strong>
- [ ] Steps to reproduce proven solved
- [ ] Test included and coverage acceptable and all Tests passing?
- [ ] PR description has layman's description of problem and solution and "Areas Affected" listed/explained?
- [ ] PR branch includes latest commits from master branch (at the time of checking)?
- [ ] Estimated points on issue
- [ ] Proper code documentation and conventions?
- [ ] Proper RTL appearance
- [ ] i18n used if applicable
- [ ] Ajax/Form used? Nonce used? Perms checked? All user input sanitized and validated? Intentional unsanitized/unescaped functions used?
- [ ] Database saved/updated? Data sanitized and validated?
- [ ] Variable output properly escaped and validated?

<strong>Peer Review</strong>
- [ ] PR branch includes latest commits from master branch (at the time of checking)?
- [ ] Steps to reproduce proven solved
- [ ] Affected areas are tested and ok
- [ ] Proper code documentation and conventions
- [ ] Proper RTL appearance
- [ ] i18n used if applicable
- [ ] Test(s) included and coverage acceptable
- [ ] Ajax/Form used? Nonce used? Perms checked? All user input sanitized and validated? Intentional unsanitized/unescaped functions used?
- [ ] Database saved/updated? Data sanitized and validated?
- [ ] Variable output properly escaped and validated?

<strong>Quality Assurance</strong>
- [ ] All tests passed on CircleCI.
- [ ] Steps to reproduce passed, the fix works.
- [ ] No errors in console or output 
- [ ] Affected areas are tested and ok
- [ ] The fix does not change a customer's website in any surprising or unintended way
  - [ ] If Divi's design <em>is</em> adjusted, settings are migrated gracefully and invisibly

<strong>Design Review</strong>
- [ ] UI is to spec
- [ ] UX is to spec
- [ ] Design is to spec
- [ ] No surprising visual changes to a customer's website will be experienced after updating
  - [ ] Any/all visual changes are migrated gracefully

<strong>Dev Review</strong>
- [ ] Test(s) included and coverage acceptable
- [ ] Ajax/Form used? Nonce used? Perms checked? All user input sanitized and validated? Intentional unsanitized/unescaped functions used?
- [ ] Database saved/updated? Data sanitized and validated?
- [ ] Variable output properly escaped and validated? 
- [ ] i18n used if applicable
- [ ] Implementation acceptable, DRY, Scaleable, future safe
- [ ] Best practices used, cleverness and grokage, ES6 use, etc.

**Comment by IvanChiurcci** [comment]
Use `custom_margin` and `custom_padding` controls instead, you can use them multiple times in a module.

**Comment by SatishKumarPrajapati** [comment]
Yes i can use that but if i use advanced margin_padding then its easy to use and custom_margin or custom_padding issue in render.

**Comment by xxtesaxx** [comment]
I believe the advanced field for margin/padding can only be added once and by default it is used for the whole module (the module wrapper). You can add as many fields of type "custom_margin" and use them for margins and paddings as you desire. It's quiet easy to implement it and you can even get mobile_options "for free". Sure it is a slight effort to implement the function to set the CSS in react and php but thats a one time thing and not too hard to do.

The best advice I can give you when doing this is to use separate top, right, bottm and left settings in ET_Builder_Element::set_style (or static css(props)) because if you just use "margin: %1$s %2$s..." and a value is not set, you end up with wrong values. Instead use "margin-top: %1$s; margin-right: %2$s;...".

**Comment by SatishKumarPrajapati** [comment]
yes i have used that but if advanced field for margin/padding can be use multiple times then it will be good and less code to write for css.

**Comment by xxtesaxx** [comment]
yes, absolutely. Ideally you could place the selector for the margin/padding inside the field declaration. That would be so much more convenient. Like

    'field' => [
        'label' => 'My Margin',
        'type' => 'custom_margin',
        'selector' => '%%order_class%% .my_selector',
        ...
    ],

**Comment by SatishKumarPrajapati** [comment]
yes @xxtesaxx and it will help to work efficiently

**Comment by jamesfosker** [comment]
> Use `custom_margin` and `custom_padding` controls instead, you can use them multiple times in a module.

How to do you get the value?
